### PR TITLE
Add "How to make a composer package" tutorial

### PR DESCRIPTION
**Pull request by tombenevides**
## Type of change
Adding a tutorial article.

### What should this PR do?
Add a tutorial about how to make a Composer package.

### Why are we making this change?
Making the website useful for a larger audience (because of more content)

### What are the acceptance criteria? 
Should be very descriptive and easy to read/follow at the same time.

### How should this PR be tested?
No need to write tests

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.

### CODE REVIEW

**Comment by ltagliaferri** [comment]
Thanks, Tom, I am going to send a contract to you via email and some other details before reviewing and merging. Thanks!

File: posts/how-to-make-a-composer-package.md [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+/>
+
+With the environment configured, we can move onto initializing our project.
+
+## Initializing the project
+
+We can now start creating our project. Let's run the following command:
+
+<Highlighter
+input='sg init'
```
Comment (ltagliaferri) [inline posts/how-to-make-a-composer-package.md:107]: I am unable to get this command to run on a clean Linux server. I added Composer to the prerequisites but this is not running as an executable, is there something else that needed to be set up first?

**Review by ltagliaferri — CHANGES_REQUESTED** [review]
Thank you @tombenevides for this tutorial, it is thorough and well explained. I have made some edits and tested it. The main issue I have encountered with the tech test is with the `sg init` command. The `sg` file is not running as an executable by Composer on a clean Linux server with the prerequisites as stated. I looked at the Composer documentation and they are running `composer init`. For Docker, the `init` command is deprecated. Can you take a look and see if there is a step or prerequisite missing and make sure you're using all the updated versions of the tools?

Thank you!

File: posts/how-to-make-a-composer-package.md (hunk 1 above)
Comment (tombenevides) [reply posts/how-to-make-a-composer-package.md:107]: Oh, sorry about that! I think I forgot to put the symbol to point to the current directory, so, the command is actually:

```bash
./sg init
```

just `sg` will run nothing cause is not a real command or alias in the bash.

image

File: posts/how-to-make-a-composer-package.md [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+ 
+ PASS  Tests\SaidQuoteTest
+  ✓ it returns a Jane Austen quote
+  ✓ it returns a message when it doesnt find a matching quote.
+  ✓ it returns a message when was not given an author name
+ 
+  Tests:  3 passed
+  Time:   0.02s`}
+language='bash'
+/>
```
Comment (ltagliaferri) [inline posts/how-to-make-a-composer-package.md:519]: I'm getting this output:

```

   Whoops\Exception\ErrorException 

  The use statement with non-compound name 'GuzzleHttpClient' has no effect

  at tests/QuoteTest.php:3
      1▕ <?php
      2▕  
  ➜   3▕ use GuzzleHttpClient;
      4▕ use GuzzleHttpHandlerMockHandler;
      5▕  
      6▕ beforeEach(function () {
      7▕     $this->mockHandler = new MockHandler();
      8▕  
      9▕     $this->mockedClient = new Client([
```

I noticed that your `use` statements are different in your `said-quotes` repo on GitHub. Do you know what the issue may be here?

Comment (tombenevides) [reply posts/how-to-make-a-composer-package.md:523]: Yes... Now that you've pointed out, all imports and namespace are without the backslashes. I don't understand why, but the current snippet in this PR is backslashed, but when I look at the preview link, it's not. I noticed that I didn't import all the classes used in the tests, so I pushed a commit to fix that. However, the error you showed me is about something I don't think I can control unless the script that converts Markdown to html ignores the backslashes. Perhaps in this case, adding an extra backslash will solve the problem, but it doesn't make much sense as the error occurs within a codeblock section, so, I can add the extra backslashes if you confirm that this is the problem.

Comment (tombenevides) [reply posts/how-to-make-a-composer-package.md:523]: image

I did a test and as you can see it works with an extra backslash. Unfortunately, it also adds a space before and after the slash. I've tried looking at the repo of the package you guys use to see if they provide an example for escaping that kind of special character, but I haven't found any examples. So if someone who knows more about the package can show us the best way to escape backslashes without adding extra spaces, that would be nice. Until then, I'll test the package later and see if I can find any information.

Comment (ltagliaferri) [reply posts/how-to-make-a-composer-package.md:523]: We will figure this out on our end, thanks for updating this!

Comment (ltagliaferri) [reply posts/how-to-make-a-composer-package.md:523]: @tombenevides — feel free to start on the next tutorial, we will get this published by resolving the markdown bug

Comment (ltagliaferri) [reply posts/how-to-make-a-composer-package.md:523]: @katjuell I created [DEVED-278](https://sourcegraph.atlassian.net/browse/DEVED-278?atlOrigin=eyJpIjoiYmY3OWUzOGU2YmU0NDRkNzlmODhiODJmNDExZjM3MWMiLCJwIjoiaiJ9) with this bug for our next sprint. I believe the regex logic is using `\` as an escape character and so in the case of 

```
namespace Tombenevides\SaidQuotes;
```

it is rendering

```
namespace TombenevidesSaidQuotes;
```

See also Tom's description above and the screenshots in the ticket. Thanks!

**Review by ltagliaferri — APPROVED** [review]
Thank you for investigating the issues, @tombenevides ! We are going to escape the backslashes by doing the double backslash (`\\`) as you have suggested, as it's at a high level of the package. Merging this in now 🚀

**Pull request merged**